distributed/cli/dask_worker.py

```diff
@@ -167,9 +186,27 @@ def f():
                 json.dump(msg, f)
         loop.add_callback(f)
 
+    retired = [False]
+
+    if lifetime:
+        @gen.coroutine
+        def retire():
+            yield gen.sleep(lifetime)
+            logger.info("Retiring worker...")
+            with rpc(ip=nannies[0].scheduler.ip,
+                     port=nannies[0].scheduler.port) as scheduler:
+                workers = ([n.worker_address for n in nannies
+                            if n.process and n.worker_port] if nanny else
+                           [n.address for n in nannies])
+                scheduler.retire_workers(workers=workers, remove=False)
+            retired[0] = True
+            logger.info("Worker retired")
+
+        loop.add_callback(retire)
```

Comment:
It would be nice to see this logic (or really any logic) within the Worker class rather than within the command line interface.  It is also *much* easier to test the worker class directly.  http://distributed.readthedocs.io/en/latest/develop.html#writing-tests